metrics/sinks/influxdb/influxdb.go

```diff
@@ -160,17 +168,32 @@ func (sink *influxdbSink) ExportData(dataBatch *core.DataBatch) {
 
 			dataPoints = append(dataPoints, point)
 			if len(dataPoints) >= maxSendBatchSize {
-				sink.sendData(dataPoints)
+				sink.wg.Add(1)
+				sink.conChan <- struct{}{}
+				go func(dataPoints []influxdb.Point) {
+					sink.sendData(dataPoints)
+				}(dataPoints)
 				dataPoints = make([]influxdb.Point, 0, 0)
 			}
 		}
 	}
 	if len(dataPoints) >= 0 {
-		sink.sendData(dataPoints)
+		sink.wg.Add(1)
+		sink.conChan <- struct{}{}
+		go func(dataPoints []influxdb.Point) {
+			sink.sendData(dataPoints)
+		}(dataPoints)
 	}
+
+	sink.wg.Wait()
 }
 
 func (sink *influxdbSink) sendData(dataPoints []influxdb.Point) {
+	defer func() {
+		<-sink.conChan
```

Comment:
and here: `// empty an item from the channel so the next waiting request can run`

Reply:
Will do.